docs/architecture/gas/README.md

```diff
@@ -32,27 +130,71 @@ action is part of a fresh transaction, or inside
 [logic.rs](https://github.com/near/nearcore/blob/14b8ae2c7465444c9b672a23b044c00be98f6e34/runtime/near-vm-logic/src/logic.rs)
 through `fn pay_action_base(..)` if the action is generated by a function call.
 The send fee is meant to cover the cost to validate an action and transmit it
-over the network. It can be different for local receipts (sender = receiver).
+over the network.
 
 The second step is action execution. It is charged in `fn apply_action(..)`.
 The execution cost has to cover everything required to apply the action to the
 blockchain's state.
 
+These two steps are done on the same shard for local receipts. For remote
+receipts, it is not clear whether they execute on the same or on different
+shards. So the cost is always charged as if it was executed on two different
```

Comment:
"each account is conceptuallya shard" is an interesting invariant we have. We _could_ actually tell if the destination account is in the same shard, we just choose not to, to avoid encouraging colocation.

Reply:
Right. Rephrased to make this invariant clear and also present the motivation of it.